hw/drivers/sensors/dps368/src/dps368_comm_interface.c

```diff
@@ -0,0 +1,325 @@
+                    itf->si_addr, reg, value);
+        STATS_INC(g_dps368_stats, read_errors);
+    }
+
+    return rc;
+}
+
+/**
+ * Writes a single byte to the specified register using SPI
+ * interface
+ *
+ * @param The sensor interface
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+static int
+dps368_spi_write_reg(struct sensor_itf *itf, uint8_t reg, uint8_t value)
+{
+    int rc;
+
+    /* Select the device */
+    hal_gpio_write(itf->si_cs_pin, 0);
+
+    /* Send the register address w/write command */
+    rc = hal_spi_tx_val(itf->si_num, reg & ~DPS368_SPI_READ_CMD_BIT);
+    if (rc == 0xFFFF) {
+        rc = SYS_EINVAL;
+        DPS368_LOG(ERROR, "SPI_%u register write failed addr:0x%02X\n",
+                    itf->si_num, reg);
+        STATS_INC(g_dps368_stats, write_errors);
+        goto err;
+    }
+
+    /* Write data */
+    rc = hal_spi_tx_val(itf->si_num, value);
+    if (rc == 0xFFFF) {
+        rc = SYS_EINVAL;
+        DPS368_LOG(ERROR, "SPI_%u write failed addr:0x%02X\n",
+                    itf->si_num, reg);
+        STATS_INC(g_dps368_stats, write_errors);
+        goto err;
+    }
+
+    rc = 0;
+
+err:
+    /* De-select the device */
+    hal_gpio_write(itf->si_cs_pin, 1);
+
+    os_time_delay((OS_TICKS_PER_SEC * 30)/1000 + 1);
+
+    return rc;
+}
+
+
+/**
+ * Read bytes from the specified register using i2c interface
+ *
+ * @param The sensor interface
+ * @param The register address to read from
+ * @param The number of bytes to read
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+static int
+dps368_i2c_read_regs(struct sensor_itf *itf, uint8_t reg, uint8_t size,
+    uint8_t *buffer)
+{
+    int rc;
+    struct hal_i2c_master_data wdata = {
+        .address = itf->si_addr,
+        .len = 1,
+        .buffer = &reg
+    };
+    struct hal_i2c_master_data rdata = {
+        .address = itf->si_addr,
+        .len = size,
+        .buffer = buffer,
+    };
+
+    rc = i2cn_master_write_read_transact(itf->si_num, &wdata, &rdata,
+                                         MYNEWT_VAL(DPS368_I2C_TIMEOUT_TICKS) * (size + 1),
+                                         1, MYNEWT_VAL(DPS368_I2C_RETRIES));
+    if (rc) {
+    	DPS368_LOG(ERROR, "I2C access failed at address 0x%02X\n",
+                    itf->si_addr);
+        STATS_INC(g_dps368_stats, read_errors);
+        return rc;
+    }
+
+    return rc;
+}
+
+
+/**
+ *
+ * Read bytes from the specified register using SPI interface
+ *
+ * @param The sensor interface
+ * @param The register address to read from
+ * @param The number of bytes to read
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+static int
+dps368_spi_read_regs(struct sensor_itf *itf, uint8_t reg, uint8_t size,

```

Comment:
This function should be wrapped with a #IF that checks if SPI is present. This file currently fails to build on systems without SPI.